src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java

```diff
@@ -71,12 +99,133 @@ public void validateLocation(final BindingResult errors, final Integer locationI
 			throw new ApiRequestValidationException(errors.getAllErrors());
 		}
 
-		final Location location = locationDataManager.getLocationByID(locationId);
+		final Location location = this.locationDataManager.getLocationByID(locationId);
 
 		if (location == null) {
 			errors.reject("location.invalid", "");
 			throw new ApiRequestValidationException(errors.getAllErrors());
 		}
 	}
 
+	public void validate(final Integer locationId, final LocationRequestDto locationRequestDto) {
```

Comment:
I think it is clear when using having 2 services exposed, one to validate updates and another for the creation instead of relying on the locationId null value.

Reply:
I will split it in two validations create and update.